File: 0001-two-sum/0001-two-sum.cpp
// O(n^2) approach:
// - nested for loop
// - compare each number to next

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); ++i) {
            for (int j = 0; j < nums.size(); ++j) {
                if (i == j)
                    j += 1;
                if ((nums[i] + nums[j]) == target)
                    return vector<int>{i, j};
            }
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = {}

        for i, v in enumerate(nums):
            diff = target - v

            if diff in prevMap:
                res = [prevMap[diff], i]
                break
                
            prevMap[v] = i

        return res
